server/src/main/java/com/cloud/template/TemplateManagerImpl.java

```diff
@@ -1178,16 +1177,28 @@ public boolean detachIso(long vmId, boolean forced) {
 
     @Override
     @ActionEvent(eventType = EventTypes.EVENT_ISO_ATTACH, eventDescription = "attaching ISO", async = true)
-    public boolean attachIso(long isoId, long vmId, boolean forced) {
+    public boolean attachIso(long isoId, long vmId, Boolean... extraParams) {
         Account caller = CallContext.current().getCallingAccount();
         Long userId = CallContext.current().getCallingUserId();
 
+        boolean forced = extraParams != null && extraParams.length > 0 ? extraParams[0] : false;
+        boolean isVirtualRouter = extraParams != null && extraParams.length > 1 ? extraParams[1] : false;
```

Comment:
add comment/javadoc to indicate the expected order of the bool extra params

Reply:
Same on this one